kong/plugins/oidc/handler.lua

```diff
@@ -124,5 +145,50 @@ function introspect(oidcConfig)
   return nil
 end
 
+function get_userinfo(oidcConfig, introspect_response)
```

Comment:
did you copy and paste this from somewhere?  if so, indicate the reference in a comment

Reply:
Only part that is somewhat copied is caching logic from here: https://github.com/zmartzone/lua-resty-openidc/blob/master/lib/resty/openidc.lua#L1575 to match introspection caching logic.